src/app-gocardless/banks/danskebank-dabno22.js

```diff
@@ -0,0 +1,58 @@
+import {
+  printIban,
+  amountToInteger,
+  sortByBookingDateOrValueDate,
+} from '../utils.js';
+
+/** @type {import('./bank.interface.js').IBank} */
+export default {
+  institutionIds: ['DANSKEBANK_DABANO22'],
+
+  normalizeAccount(account) {
+    return {
+      account_id: account.id,
+      institution: account.institution,
+      mask: account.iban.slice(-4),
+      iban: account.iban,
+      name: [account.name, printIban(account)].join(' '),
+      official_name: account.name,
+      type: 'checking',
+    };
+  },
+
+  normalizeTransaction(transaction, _booked) {
+    /**
+     * The valueDate in transactions from Danske Bank is not the one expected, but rather the date
+     * the funds are expected to be paid back for credit accounts.
+     */
+    delete transaction.valueDate;
+
+    /**
+     * Danske Bank seems to have a bug in their API where it appends the EndToEndID to
+     * remittanceInformationUnstructured, cluttering the data.
+     *
+     * We clean thais up by removing any instances of this string from all transactions.
+     *
+     */
+    transaction.remittanceInformationUnstructured =
+      transaction.remittanceInformationUnstructured.replace(
+        '\nEndToEndID: NOTPROVIDED',
+        '',
+      );
+    return transaction;
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏
```suggestion
    return {
      ...transaction,
      date: transaction.bookingDate,
    };
```

and then you can also remove line 28